### ISSUE TITLE
Single Sign-On

### ISSUE THREAD

**Issue by erviker** [TwiN/gatus#158]
SSO as an authentication option would be an awsome edtion!

**Comment by erviker** [comment]
config exsampel
```
sso:
     type: oauth2
     provider: google
     id:
     secret:
```

**Comment by erviker** [comment]
Expanding on this having public also having private checks that you get access to after you login

**Comment by TwiN** [comment]
FYI I started working on support for OIDC in #228

**Comment by TwiN** [comment]
Closed by #228

**Comment by TwiN** [comment]
Released in [v3.5.0](https://github.com/TwiN/gatus/releases/tag/v3.5.0)
